compiler/src/openxla/compiler/nvgpu/Transforms/ConvertMHLOToCUDNN.cpp

```diff
@@ -23,72 +24,144 @@
 
 using namespace mlir;
 
-namespace openxla::compiler::nvgpu {
+namespace openxla::compiler::nvgpu::cudnn {
 
-static SmallVector<int64_t> getRowMajorStrides(ArrayRef<int64_t> shape) {
-  SmallVector<int64_t> strides(shape.size(), 1);
-  std::partial_sum(shape.rbegin(),
-                   shape.rend() - 1,
-                   strides.rbegin() + 1,
-                   std::multiplies<int64_t>());
-  return strides;
+static TensorType getCudnnTensorType(mlir::TensorType tensor_type) {
+  return TensorType::get(tensor_type.getShape(), tensor_type.getElementType());
 }
 
-static cudnn::TensorDescType getTensorDescType(TensorType tensor_type) {
-  auto shape = tensor_type.getShape();
-  Type element_type = tensor_type.getElementType();
-  int alignment = 0;
-  auto strides = getRowMajorStrides(shape);
-  return cudnn::TensorDescType::get(tensor_type.getContext(),
-                                    shape, element_type, alignment, strides);
-}
+// Outlines all transitive ops defining 'result' into a cudnn.graph op to be
+// called with `arguments`.
+static LogicalResult outlineToGraph(
+    TypedValue<mlir::TensorType> result,
```

Comment:
Should we rename `cudnn::TensorType` to `CuDNNTensorType` to avoid lookup collisions with a regular tensor type?

Reply:
I think that would be cleaner, yes.

I'm not fully convinced yet about the casing though. Shouldn't this rather be `CudnnTensorType`?

- `cuDNN`: in comments, referring to the library name.
- `cudnn`: lower case, e.g. for variable names.
- `Cudnn`: Camel Case, e.g. for type names.
- `CUDNN`: UPPER CASE, e.g. in file names.
- ~~CuDNN~~: no need.

Reply:
`Cudnn` in type names SGTM, will gradually send PRs cleaning up names everywhere. I'd also do something about file/folder names, CUDNN is a bit too much upper case letters to my taste. We should add naming convention for all NVIDIA libraries we want to add to readme